lib/kubeclient/common.rb

```diff
@@ -39,6 +39,12 @@ module ClientMixin
       'fieldSelector' => :field_selector
     }.freeze
 
+    DELETE_ARGUMENTS = {
+      'gracePeriodSeconds' => :grace_period_seconds,
+      'orphanDependents'   => :orphan_dependents,
```

Comment:
I suggest to focus on the non deprecated param, unless there's a big demand for prior versions.
However it also depends on when propagation policy was introduced, on other words, will using it cause errors on earlier k8s versions?
@moolitayer @simon3z wdyt?